openapi.yaml

```diff
@@ -0,0 +1,1252 @@
+                    description: The total number of results
+                  results:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/unit'
+  /unit/{id}/:
+    get:
+      tags:
+      - unit
+      description: The unit endpoint returns units (libraries, youth
+        centers etc.) listed in the reservation system.
+      parameters:
+      - name: id
+        in: path
+        description: Unique identifier for the unit in the service registry.
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/unit'
+  /purpose/:
+    get:
+      tags:
+      - filter
+      description: The purpose endpoint returns the possible resource usage purposes registered in the system. 
+        The optional parameter **page** allows specifying page number and **page_size** allows specifying more than the default 
+        50 purposes per page.
+      parameters:
+      - name: page
+        in: query
+        description: Result page number
+        schema:
+          type: integer
+      - name: page_size
+        in: query
+        description: Number of purposes per page
+        schema:
+          type: integer
+      responses:
+        200:
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  next:
+                    type: string
+                    description: URL for the next page
+                  prev:
+                    type: string
+                    description: URL for the previous page
+                  count:
+                    type: integer
+                    description: The total number of results
+                  results:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/purpose'
+  /purpose/{id}/:
+    get:
+      tags:
+      - filter
+      description: The purpose endpoint returns the possible resource usage purposes
+        registered in the system.
+      parameters:
+      - name: id
+        in: path
+        description: Unique identifier of the usage purpose.
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/purpose'
+  /type/:
+    get:
+      tags:
+      - filter
+      description: The type endpoint returns the possible resource types registered in the system. The optional parameter **page** 
+      allows specifying page number and **page_size** allows specifying more than the default 20 units per page.
+      parameters:
```

Comment:
Validator seems to have a problem with line 198: `Can not read a block mapping entry; a multiline key may not be an implicit key`

Reply:
Fixed by adding spaces. Validator seems happy now.